### DISCUSSION TITLE
How to define shared library packages between apps

### DISCUSSION THREAD

**Discussion by tunahan-guler** [web-infra-dev/modern.js#4710] [Q&A]
How will I define that the common packages used by the main application and the sub-application are shared packages?

**Comment by chenjiahan** [comment]
A common practice is to create a new package in the monorepo and reference this package in the app through pnpm's workspace protocol.

- directory:

```bash
/packages
  /app1
  /app2
  /lib1
  /lib2
```

- app1/package.json:

```json
{
  "dependencies": {
    "lib1": "workspace:*"
  }
}
```

Ref: https://pnpm.io/workspaces
